tests/infrastructure/src/main/java/org/mule/test/infrastructure/profiling/SpanTestHierarchy.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.test.infrastructure.profiling;
+
+import static org.hamcrest.Matchers.notNullValue;
+import static org.junit.Assert.assertThat;
+
+import org.mule.runtime.core.privileged.profiling.CapturedExportedSpan;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+
+public class SpanTestHierarchy {
+
+  private SpanNode root;
+  private SpanNode currentNode;
+  private SpanNode lastChild;
+  private final String NO_PARENT_SPAN = "0000000000000000";
+  private HashSet<String> visitedSpans = new HashSet();
+  private HashMap<String, CapturedExportedSpan> spanHashMap = new HashMap<>();
+  private Collection<CapturedExportedSpan> actualExportedSpans;
+
+  public SpanTestHierarchy(Collection<CapturedExportedSpan> actualExportedSpans) {
+    this.actualExportedSpans = actualExportedSpans;
+    actualExportedSpans.forEach(span -> spanHashMap.put(span.getSpanId(), span));
+  }
+
+  public SpanTestHierarchy withRoot(String rootName) {
+    root = new SpanNode(rootName);
+    root.parent = new SpanNode(NO_PARENT_SPAN);
+    currentNode = root;
+    return this;
+  }
+
+  public SpanTestHierarchy beginChildren() {
+    lastChild = currentNode;
+    return this;
+  }
+
+  public SpanTestHierarchy child(String childName) {
+    SpanNode child = new SpanNode(childName);
+    child.parent = lastChild;
+    lastChild.addChild(child);
+    currentNode = child;
+    return this;
+  }
+
+  public SpanTestHierarchy endChildren() {
+    if (currentNode.parent != null) {
+      lastChild = lastChild.parent;
+      currentNode = currentNode.parent.parent;
+    }
+    return this;
+  }
+
+  public SpanNode getRoot() {
+    return root;
+  }
+
+  public void assertSpanTree(SpanNode node, CapturedExportedSpan parent) {
```

Comment:
add jdoc to public methods.

Reply:
Do I really need the parent parameter in the public method or is it only needed to perform the recursion?I would any way create a public method where you don't pass the second parameter so that you don't need to pass null (which is the most common use).